js-api-spec/node-package-importer.node.test.ts

```diff
@@ -0,0 +1,681 @@
+// Copyright 2023 Google Inc. Use of this source code is governed by an
+// MIT-style license that can be found in the LICENSE file or at
+// https://opensource.org/licenses/MIT.
+
+import {
+  compile,
+  compileAsync,
+  compileString,
+  compileStringAsync,
+  render,
+  renderSync,
+  nodePackageImporter,
+  NodePackageImporter,
+  LegacyException,
+  LegacyResult,
+  FileImporter,
+} from 'sass';
+
+import {sandbox, SandboxDirectory} from './sandbox';
+
+const fileImporter = (dir: SandboxDirectory): FileImporter => ({
+  findFileUrl: file => dir.url(file),
+});
+
+function manifestBuilder(overrides?: {[key: string]: string | Object}) {
+  return JSON.stringify({
+    name: 'foo',
+    version: '1.0.0',
+    description: 'Verifying proposed pkg: loading algorithm',
+    scripts: {
+      test: 'echo "Error: no test specified" && exit 1',
+    },
+    author: 'Package Author',
+    license: 'ISC',
+    ...overrides,
+  });
+}
+
+function exportsBuilder(key: string) {
+  return {
+    exports: {
+      '.': {
+        [key]: './src/sass/_styles.scss',
+      },
+      './colors/index.scss': {
+        [key]: './src/sass/colors/index.scss',
+      },
+      './_variables.scss': {
+        [key]: './src/sass/_variables.scss',
+      },
+    },
```

Comment:
It feels weird to me that this is exporting files that don't necessarily exist in all (or most) tests. I think it might be clearer to just write the necessary exports by hand. Less indirection can make the tests easier to understand as well.